packages/target-decisioning-engine/src/decisionProvider.js

```diff
@@ -58,7 +58,11 @@ function DecisionProvider(
   const { sendNotificationFunc, telemetryEnabled = true } = config;
 
   const visitorId = request.id;
-  const processRule = ruleEvaluator(clientId, visitorId);
+  const processRule = ruleEvaluator(
+    clientId,
+    visitorId,
+    config.parseDomainImpl
```

Comment:
Instead of passing in config.parseDomainImpl here I suggest making a new method that will return the implementation from the config or the basic one.  That way we are guaranteed a valid `ParseDomainFunc` every time.

For example...

decisioning-engine/utils.js
```js
/**
 *
 * @param {import("../types/DecisioningConfig").DecisioningConfig} config
 * @returns {import("../types/DecisioningContext").ParseDomainFunc}
 */
export function getParseDomainImpl(config) {
  return isFunction(config.parseDomainImpl)
    ? config.parseDomainImpl
    : parseDomainBasic;
}
```


decisionProvider.js
```js
  const processRule = ruleEvaluator(
    clientId,
    visitorId,
    getParseDomainImpl(config)
  );
```